cns/cniconflist/generator_linux_test.go

```diff
@@ -0,0 +1,77 @@
+package cniconflist_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/cni"
+	"github.com/Azure/azure-container-networking/cns/cniconflist"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/stretchr/testify/assert"
+)
+
+type bufferWriteCloser struct {
+	*bytes.Buffer
+}
+
+func (b *bufferWriteCloser) Close() error {
+	return nil
+}
+
+func TestGenerateConflist(t *testing.T) {
+	generatorTests := []struct {
+		s                       configuration.CNIConflistScenario
+		fixture                 string
+		expectedNewGeneratorErr bool
+	}{
+		{
+			s:       configuration.CNIConflistScenarioV4Overlay,
+			fixture: "testdata/fixtures/azure-linux-swift-overlay.conflist",
+		},
+		{
+			s:                       configuration.CNIConflistScenario("unknown"),
+			expectedNewGeneratorErr: true,
+		},
+	}
+
+	for _, test := range generatorTests {
+		tt := test
+		t.Run(fmt.Sprintf("generate-%s", tt.s), func(t *testing.T) {
+			buffer := new(bytes.Buffer)
+			g, err := cniconflist.NewGenerator(&bufferWriteCloser{buffer}, tt.s)
+			if err != nil {
+				if tt.expectedNewGeneratorErr {
+					return
+				}
+				t.Fail()
+			}
+
+			err = g.Generate()
+			assert.NoError(t, err)
+
+			// by unmarshalling the json and comparing the structs, we're only checking
+			// for data equality and not really diffing the files, for example
+			// the fixture could have spaces and the generated conflist could have tabs,
+			// but this is fine since all we really care about is the data equality
+			fixture, err := os.ReadFile(tt.fixture)
+			if err != nil {
+				t.Fail()
```

Comment:
I'd recomment using `t.Error` / `t.Fatal`, or their respective `require` / `assert` equivalents to make sure the failure messages are useful.

Reply:
fixed to use assert